code/model/UserDefinedForm.php

```diff
@@ -1377,9 +1377,30 @@ public function edit($request) {
 	 */
 	public function preview() {
 		return $this->customise(new ArrayData(array(
-			'Body' => $this->record->getEmailBodyContent()
+			'Body' => $this->record->getEmailBodyContent(),
+			'HideFormData' => $this->record->HideFormData,
+			'Fields' => $this->getPreviewFieldData()
 		)))->renderWith($this->record->EmailTemplate);
 	}
+
+	/**
+	 * Get some placeholder field values to display in the preview
+	 * @return ArrayList
+	 */
+	private function getPreviewFieldData() {
+		$data = new ArrayList();
+
+		foreach ($this->record->Form()->Fields() as $field) {
```

Comment:
Oh right, for some reason I thought $record was the `UserDefinedForm` instance, so all good. :D